### PR TITLE
Add FIPS guide

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Details how to build Contour and Envoy with Boring Crypto and how to
configure Contour with FIPS approved ciphers

Fixes https://github.com/projectcontour/contour/issues/2878

### CODE REVIEW

File: site/_guides/fips.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+title: FIPS
```
Comment (sunjayBhatia) [inline site/_guides/fips.md:2]: This isn't the greatest title tbh

File: site/_guides/fips.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+For a fully FIPS compliant deployment of Contour a few things are required:
+- Contour must be compiled with a FIPS validated cryptographic module
+- Envoy must be compiled with a FIPS validated cryptographic module
+- Contour must be configured to use FIPS approved cryptographic algorithms
+
+This guide will run through an example of the process for building and configuring Contour and Envoy for adherence to FIPS 140-2.
+Specifically, we will show how Contour and Envoy can be built with the FIPS validated BoringCrypto module of BoringSSL and configured to use FIPS approved TLS ciphers.
+
+Please note that this guide makes no guarantees about Contour FIPS 140-2 approval, validation, or the like.
```
Comment (sunjayBhatia) [inline site/_guides/fips.md:20]: This disclaimer could probably use some work

File: site/_guides/fips.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
+---
+title: FIPS
+layout: page
+---
+
+The [Federal Information Processing Standard (FIPS) 140-2](https://csrc.nist.gov/publications/detail/fips/140/2/final) publication describes United States government approved security requirements for cryptographic modules.
```
Comment (sunjayBhatia) [inline site/_guides/fips.md:6]: TODO: Move the links to the enumerated list at the bottom of the document once we get a pass and determine these links are good

File: site/_guides/fips.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+In this section we will describe how the [`projectcontour/contour`](https://hub.docker.com/r/projectcontour/contour) container image can be compiled and linked to BoringCrypto for FIPS compliance.
+We will be modifying the standard build process by setting up some dependencies and passing additional arguments to the same `make` target used to build the standard, non-FIPS image distributed by the project.
+
+You will need some software downloaded and installed on the computer you are performing the Contour FIPS build on:
+- Contour source code checked out to the version you would like to build
+- [GNU Make](https://www.gnu.org/software/make/)
+- [Docker](https://www.docker.com/)
+
+The Contour [Dockerfile](../../Dockerfile) uses a multistage build that performs compilation in an image that contains the necessary build tools and dependencies and then exports compiled artifacts to a final image.
```
Comment (sunjayBhatia) [inline site/_guides/fips.md:46]: TODO: Fix this

**Review by skriss — COMMENTED** [review]
This looks great, no comments as of right now. I'll try to run through a full test later this week.

**Comment by sunjayBhatia** [comment]
Siteproof failure is just b/c it's a new page

**Review by youngnick — APPROVED** [review]
LGTM, nice work.

**Review by stevesloka — APPROVED** [review]

**Comment by sunjayBhatia** [comment]
Page preview here: https://deploy-preview-3446--projectcontour.netlify.app/guides/fips/

**Comment by skriss** [comment]
> @skriss anything you want to check on before merging this?

Nah, ship it!

**Pull request merged**